### PR TITLE
Log ServiceHub process exit code on failure

### PR DESCRIPTION
**Pull request by tmat**
Includes exit code in gold bar message if RPC fails.

### CODE REVIEW

File: src/VisualStudio/Core/Def/Telemetry/VisualStudioWorkspaceTelemetryService.cs [hunk 1]
```diff
@@ -60,10 +62,46 @@ protected override void TelemetrySessionInitialized()
...
+
+                HookRemoteProcessExitCodeLogger(remoteProcessId);
             });
         }
+
+        private static void HookRemoteProcessExitCodeLogger(Optional<int> remoteProcessId)
+        {
+            if (!remoteProcessId.HasValue)
+            {
+                FaultReporter.LogError($"Roslyn telemetry service initialization failed.");
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Telemetry/VisualStudioWorkspaceTelemetryService.cs:77]: where do tehse get reported to us through?  (curious how there's no FunctionId for this).

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/VisualStudio/Core/Def/Telemetry/VisualStudioWorkspaceTelemetryService.cs (hunk 1 above)
Comment (tmat) [reply src/VisualStudio/Core/Def/Telemetry/VisualStudioWorkspaceTelemetryService.cs:77]: ServiceHub log.

**Comment by tmat** [comment]
Hmm, looks like the handler is not called when the process dies of Stack Overflow :(

**Comment by CyrusNajmabadi** [comment]
That's crazy... why would the handler, in anotehr process, be affected by SO... so weird :(

**Review by arkalyanms — APPROVED** [review]

**Comment by tmat** [comment]
@CyrusNajmabadi Changed implementation and tested that the exit code is reported on SO.

**Review by arkalyanms — APPROVED** [review]

**Pull request merged**